## Importing Requried Libraries

In [1]:
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
import requests 
#importing BeautifulSoup to handle HTML Files
!conda install -c conda-forge beautifulsoup4 --yes
from bs4 import BeautifulSoup



!pip install lxml
from lxml import html

Solving environment: done

## Package Plan ##

  environment location: /home/jupyterlab/conda/envs/python

  added / updated specs:
    - beautifulsoup4


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    beautifulsoup4-4.9.0       |   py36h9f0ad1d_0         160 KB  conda-forge
    soupsieve-1.9.4            |   py36h9f0ad1d_1          58 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         218 KB

The following NEW packages will be INSTALLED:

  beautifulsoup4     conda-forge/linux-64::beautifulsoup4-4.9.0-py36h9f0ad1d_0
  soupsieve          conda-forge/linux-64::soupsieve-1.9.4-py36h9f0ad1d_1



soupsieve-1.9.4      | 58 KB     | ##################################### | 100% 
beautifulsoup4-4.9.0 | 160 KB    | ##################################### | 100% 
Preparing transaction: done
Verifying transaction

## Opening file using request Library and assigning it to a varribale

In [2]:
#opening the file using Library and then assigning it to a varriable.

url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'

r = requests.get(url)
content =r.content

## Converting HTML format to BeautifulSoup Tree Format

In [3]:

soup = BeautifulSoup(content,'html.parser') 

## Finding Relevent data and Assigning them to Lists

In [4]:
A = []
B = []
C = []


for row in soup.findAll('tr'):
    cells = row.findAll('td')  
    
    
    if len(cells) == 3:
        if cells[1].find(text=True).rstrip()=='Not assigned':
            continue
        else :
            A.append(cells[0].find(text=True).rstrip())
            B.append(cells[1].find(text=True).rstrip())
            C.append(cells[2].find(text=True).rstrip())



## Assigning Lists into Columns

In [16]:

df = pd.DataFrame(A,columns = ['postal_code'])
df['Brough'] = B
df['Neighborhood'] = C

df

postal_code            Brough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                          Neighborhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                           Regent Park / Harbourfront  
3                    Lawrence Manor / Lawrence Heights  
4         Queen's Park / Ontario Provincial Government  
..                                                 ...  
98    The Kingsway / Montgomery Road  / Old Mill North  
99                                Church and Wellesley  
100              Business reply mail Processing CentrE  
101  Old Mill South / King's Mill Park / Sunnylea /...  
102  Mimico NW / The Queensway West / South of Bloo...  

[103 rows x 3 columns]

## Replacing '/' value with ',' in Neighborhoods Column

In [17]:
df['Neighborhood'] = df['Neighborhood'].str.replace('/' ,',')
df

postal_code            Brough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                          Neighborhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                           Regent Park , Harbourfront  
3                    Lawrence Manor , Lawrence Heights  
4         Queen's Park , Ontario Provincial Government  
..                                                 ...  
98    The Kingsway , Montgomery Road  , Old Mill North  
99                                Church and Wellesley  
100              Business reply mail Processing CentrE  
101  Old Mill South , King's Mill Park , Sunnylea ,...  
102  Mimico NW , The Queensway West , South of Bloo...  

[103 rows x 3 columns]

## Finding the shape of the Dataframe

In [7]:
df.shape

(103, 3)

##### Dataframe has 103 rows and 3 columns

## Importing CSV file with GPS Coordinates

In [32]:
df_Lat_long = pd.read_csv("Long_Lat.csv")
df_Lat_long

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
..          ...        ...        ...
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

[103 rows x 3 columns]

## Assigning Longitude and Latitude to the Original Dataframe

In [33]:
df["Latitude"] = df_Lat_long["Latitude"]
df["Longitude"] = df_Lat_long["Longitude"]
df

postal_code            Brough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                          Neighborhood   Latitude  Longitude  
0                                            Parkwoods  43.806686 -79.194353  
1                                     Victoria Village  43.784535 -79.160497  
2                           Regent Park , Harbourfront  43.763573 -79.188711  
3                    Lawrence Manor , Lawrence Heights  43.770992 -79.216917  
4         Queen's Park , Ontario Provincial Government  43.773136 -79.239476  
..                                                 ...        ...        ...  
98    The Kingsway , Montgomery Road  , Old Mill North  43.706876 -79.518188  
99                                Church and Wellesley  43.696319 -79.532242  
100              Business reply mail Processing CentrE  43.688905 -79.554724  
101  Old Mill South , King's Mill Park , Sunnylea ,...  43.739416 -79.588437  
102  Mimico NW , The Queensway West , South of Bloo...  43.706748 -79.594054  

[103 rows x 5 columns]